src/theory/partition_generator.h

```diff
@@ -0,0 +1,124 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * PartitionGenerator for creating partitions.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__SPLITTER_H
+#define CVC5__THEORY__SPLITTER_H
+
+#include <vector>
+
+#include "proof/trust_node.h"
+#include "smt/env_obj.h"
+#include "theory/theory.h"
+#include "theory/valuation.h"
+
+namespace cvc5 {
+
+class TheoryEngine;
+
+namespace prop {
+class PropEngine;
+}
+
+namespace theory {
+
+class PartitionGenerator : protected EnvObj
+{
+ public:
+  PartitionGenerator(Env& env,
+                     TheoryEngine* theoryEngine,
+                     prop::PropEngine* propEngine);
+
+  /**
+   * Make partitions for parallel solving. e communicates the effort at which makePartitions was called.
+   * Returns a lemma blocking off the emitted cube from the search.
+   */
+  TrustNode makePartitions(Theory::Effort e);
+
```

Comment:
It would be nice to add an interface for getting the cubes from this class.  `std::vector<Node> getPartition() const`.

I have a use case in mind where another internal module may want to interact with this class.